test/Microsoft.Health.SqlServer.Tests.Integration/Features/Schema/RetryTests.cs

```diff
@@ -0,0 +1,199 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Data;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Sdk.Sfc;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Health.SqlServer.Tests.Integration.Features.Schema;
+
+#pragma warning disable CA2007 // Consider calling ConfigureAwait on the awaited task
+public class RetryTests : SqlIntegrationTestBase
+{
+    public RetryTests(ITestOutputHelper outputHelper)
+        : base(outputHelper)
+    {
+    }
+
+    private const int RetryError = 8134; // divide by zero
+
+    [Fact]
+    public async Task RetriesBehavior()
+    {
+        try
+        {
+            await CreateTestTable();
+
+            // For non query retries are handled by command retry
+            await CreateTestStoredProcedureWithErrorBeforeSelect();
+            Assert.False(await ErrorIsLogged());
+            await ExecuteNonQuery();
+            Assert.True(await ErrorIsLogged());
+
+            // if error happens before select it is handled by command retry
+            // same stored procedure as above just reader added
+            await ResetError();
+            Assert.False(await ErrorIsLogged());
+            var results = await ExecuteQuery();
+            Assert.True(await ErrorIsLogged());
+            Assert.Equal(10, results.Count);
+
+            await ResetError();
+            Assert.False(await ErrorIsLogged());
+            // if error happens in select it is not handled
```

Comment:
Is it because the transient error a different one than the default list?

Reply:
See full answer on other comment.